packages/backend/src/apps/spotify/common/add-auth-header.ts

```diff
@@ -0,0 +1,12 @@
+import { TBeforeRequest } from '@automatisch/types';
+
+const addAuthHeader: TBeforeRequest = ($, requestConfig) => {
+  if ($.auth.data?.accessToken) {
```

Comment:
As this function is used as a middleware before HTTP requests, the current implementation overrides the existing Authorization header, as @zntemel mentioned in the PR description. To cover this scenario, I'd suggest extending this condition to cover the existing Authorization headers.

```suggestion
  if (!requestConfig.headers.Authorization && $.auth.data?.accessToken) {
```